core/src/main/java/org/apache/accumulo/core/metrics/MetricsDocGen.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.metrics;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.TreeMap;
+
+/**
+ * This class generates documentation to inform users of the available metrics in a presentable
+ * form.
+ */
+public class MetricsDocGen {
+  private final PrintStream doc;
+  private final TreeMap<String,Metric> sortedMetrics = new TreeMap<>();
```

Comment:
Since you don't need the name, and are just using it for sorting, you can make this more clear with an explicit comparator.

```suggestion
  private final TreeSet<Metric> sortedMetrics = new TreeSet<>(Comparator.comparing(Metric::getName));
```